src/assets/scripts/client/measurement/MeasureTool.js

```diff
@@ -0,0 +1,363 @@
+        // as the first point in the path
+        // We also calculate the distance and time take to fly the turn
+        if (initialTurn !== null) {
+            const { exitPoint, arcLength } = initialTurn;
+            const distance = nm(arcLength);
+            const duration = Math.round((distance / groundSpeed) * TIME.ONE_HOUR_IN_SECONDS);
+
+            initialValues.lastPoint = exitPoint;
+            initialValues.totalDistance = distance;
+            initialValues.totalDuration = duration;
+
+            pointsList[0] = exitPoint;
+        }
+
+        // The first leg is just an end point
+        const firstPoint = this._getRelativePosition(pointsList.shift());
+        const firstLeg = new MeasureLegModel(firstPoint);
+        initialValues.previousLeg = firstLeg;
+
+        pointsList.reduce((lastValues, point) => {
+            const { previousLeg } = lastValues;
+            let { totalDistance, totalDuration } = lastValues;
+            const leg = new MeasureLegModel(
+                this._getRelativePosition(point),
+                turnRadius,
+                previousLeg
+            );
+            const distance = nm(leg.distance);
+            const bearing = heading_to_string(leg.bearing);
+            let duration = 0;
+
+            if (groundSpeed !== null) {
+                duration = Math.round((distance / groundSpeed) * TIME.ONE_HOUR_IN_SECONDS);
+            }
+
+            const labels = [
+                this._getLabel(distance, duration, bearing)
+            ];
+
+            totalDistance += distance;
+            totalDuration += duration;
+
+            if (totalDistance !== distance) {
+                labels.push(this._getLabel(totalDistance, totalDuration));
+            }
+
+            leg.labels = labels;
+
+            return {
+                previousLeg: leg,
+                totalDistance,
+                totalDuration
+            };
+        }, initialValues);
+
+        return {
+            initialTurn,
+            firstLeg
+        };
+    }
+
+    /**
+     * Removes the last point from the path
+     *
+     * @for MeasureTool
+     * @method removeLastPoint
+     */
+    removeLastPoint() {
+        // The "last" point is actually the 2nd last item in _points
+        // This means the path will end at the cursor position when redrawn
+        const { length } = this._points;
+        if (length > 2) {
+            this._points.splice(length - 2, 1);
+        }
+    }
+
+    /**
+     * Starts the measure tools for the specifeid aircraft
+     * and adds an initial leg
+     *
+     * The startValue can be an object with a `#relativePosition` property,
+     * or a array of numbers that represent a `relativePosition`
+     *
+     * @for MeasureTool
+     * @method startMeasuring
+     * @param startValue {AircraftModel|FixModel|array<number>}
+     */
+    startMeasuring(startValue) {
+        this._points = [startValue];
```

Comment:
is `_points` an empty array before this assignment?  is so, you should use `.push()` here.  this is actually mutating `_points` instead of updating

Reply:
Yes it should always be empty, but it was my way of making sure it was (which is a ~good~ bad way of ensuring bugs never manifest themselves)...